src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs

```diff
@@ -156,6 +160,25 @@ public SplashScreen(bool enableSignInButton = true)
             this.enableSignInButton = enableSignInButton;
         }
 
+        protected override void OnClosing(CancelEventArgs e)
+        {
+            // If we have multiple OnClosing events (ex Clicking the close button multiple times)
+            // we need to only process the first one. THe rest should be canceled so that we can avoid timing issues with the order of windows messages
+            // Ex  WM_CLOSE => webview2.Visibility.Set => waits for windows message =>  WM_DESTROY =>
+            // webview2.Dispose => webview2.Visible.Set receives windows message => crash because object got disposed. 
+            if (!IsClosing)
+            {
+                // First call to OnClosing
+                IsClosing = true;
+            }
+            else
+            {
+                // Cancel the Close action for all subsequent calls
+                e.Cancel = true;
```

Comment:
Setting `e.Cancel = true` will cancel the close event (i.e CLOSE events will stop cascading further)
There is a bit of risk with this approach because setting `e.Cancel = true` causes some internal flags to be set too (ex _isClosing will be set to false) even though the first OnClosing call was not canceled and might still be waiting for messages to come through.
https://github.com/dotnet/wpf/blob/0abbfc749562e3428a11b7dbbc33ca2c30868194/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Window.cs#L4244C17

Reply:
so this situation and override are only required for windows that contain webview2 instances?

Reply:
At least id Dynamo, I think it is.